crates/wit-parser/src/ast/resolve.rs

```diff
@@ -4,14 +4,18 @@ use anyhow::Result;
 use indexmap::IndexMap;
 use std::collections::{HashMap, HashSet};
 use std::mem;
+use topo_sort::{TopoSort, SortResults};
 
 #[derive(Default)]
 pub struct Resolver {
+    // Mapping of (interface-name, type-name) to type-id for resolving use.
+    qualified_type_names: HashMap<(String, String), TypeId>,
```

Comment:
Instead of a two-level hashmap like this could this perhaps be a `Vec<HashMap<String, TypeId>>` where the position within the vector corresponds to the `InterfaceId` index and within that is the namespace of names within that interface?